pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -158,26 +117,82 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 
 	ec2svc := ec2.NewService(scope.Scope)
 
-	clusterMachines, err := scope.MachineClient.List(v1.ListOptions{})
+	log.Info("Retrieving machines for cluster")
+	clusterMachines, err := scope.MachineClient.List(actuators.ListOptionsForCluster(cluster.Name))
 	if err != nil {
 		return errors.Wrapf(err, "failed to retrieve machines in cluster %q", cluster.Name)
 	}
 
 	controlPlaneMachines := GetControlPlaneMachines(clusterMachines)
+	if len(controlPlaneMachines) == 0 {
+		log.Info("No control plane machines exist yet - requeuing")
+		return &controllerError.RequeueAfterError{RequeueAfter: 5 * time.Second}
+	}
+
+	// Except for the very first control plane machine, everything will be a join, so let's assume we're joining.
+	isNodeJoin := true
+	if cluster.Annotations[v1alpha1.AnnotationControlPlaneReady] != "true" {
+		if machine.Spec.Versions.ControlPlane == "" {
+			// This isn't a control plane machine - have to wait
+			log.Info("No control plane machines exist yet - requeuing")
+			return &controllerError.RequeueAfterError{RequeueAfter: 5 * time.Second}
+		}
 
-	isNodeJoin, err := a.isNodeJoin(scope, controlPlaneMachines)
-	if err != nil {
-		return errors.Wrapf(err, "failed to determine whether machine %q should join cluster %q", machine.Name, cluster.Name)
+		configMapName := actuators.ControlPlaneConfigMapName(cluster)
+		log.Info("Checking for existence of control plane configmap lock", "configmap-name", configMapName)
+
+		controlPlaneConfigMap, err := a.coreClient.ConfigMaps(cluster.Namespace).Get(configMapName, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			controlPlaneConfigMap = &apicorev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: cluster.Namespace,
+					Name:      configMapName,
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: cluster.APIVersion,
+							Kind:       cluster.Kind,
+							Name:       cluster.Name,
+							UID:        cluster.UID,
+						},
+					},
+				},
+			}
+
+			log.Info("Attempting to create control plane configmap lock")
+			_, err = a.coreClient.ConfigMaps(cluster.Namespace).Create(controlPlaneConfigMap)
```

Comment:
Do we need the Get before this Create call?

Reply:
The Get is on line 144. If we can have Get return a successful response code, that is a good thing. We had some Velero users complaining because it always tried to create a resource, and most of the time it would fail (already exists). It shows up in the GCP/GKE error logs and they didn't like that.

Reply:
Fair enough!